types/types.go

```diff
@@ -263,6 +263,19 @@ func (s ServiceConfig) GetDependencies() []string {
 	return dependencies
 }
 
+// GetDependent retrieves all services which depend on this service
+func (s ServiceConfig) GetDependent(p *Project) []string {
```

Comment:
Nit:
```suggestion
func (s ServiceConfig) GetDependents(p *Project) []string {
```

Also, I wonder if passing `*Project` back in here is a bit odd (though having this next to `GetDependencies()` is nice).

Do you think `Project::GetDependentsForService(name string) []string` would make more sense?